api/src/main/kotlin/gropius/schema/mutation/ArchitectureMutations.kt

```diff
@@ -0,0 +1,483 @@
+            dfe.gropiusAuthorizationContext, input
+        )
+    }
+
+    @GraphQLDescription(
+        """Deletes the specified IMSProject, requirse ADMIN on the IMS associted with the
+        specified IMSProject OR MANAGE_IMS on the Trackable associated with the specified
+        IMSProject.
+        """
+    )
+    @AutoPayloadType("The id of the deleted IMSProject")
+    suspend fun deleteIMSProject(
+        @GraphQLDescription("Defines which IMSProject to delete")
+        input: DeleteNodeInput, dfe: DataFetchingEnvironment
+    ): ID {
+        imsProjectService.deleteIMSProject(dfe.gropiusAuthorizationContext, input)
+        return input.id
+    }
+
+    @GraphQLDescription(
+        """Creates a new IntraComponentDependencySpecification, requires ADMIN on the Component associated with the 
+        specified ComponentVersion.
+        """
+    )
+    @AutoPayloadType("The created IntraComponentDependencySpecification")
+    suspend fun createIntraComponentDependencySpecification(
+        @GraphQLDescription("Defines the created IntraComponentDependencySpecification")
+        input: CreateIntraComponentDependencySpecificationInput, dfe: DataFetchingEnvironment
+    ): IntraComponentDependencySpecification {
+        return intraComponentDependencySpecificationService.createIntraComponentDependencySpecification(
+            dfe.gropiusAuthorizationContext, input
+        )
+    }
+
+    @GraphQLDescription(
+        """Updates the specified IntraComponentDependencySpecification, requires ADMIN on the Component associated with the 
+        IntraComponentDependencySpecification to update."""
+    )
+    @AutoPayloadType("The updated IntraComponentDependencySpecification")
+    suspend fun updateIntraComponentDependencySpecification(
+        @GraphQLDescription("Defines which IntraComponentDependencySpecification to update and how to update it")
+        input: UpdateIntraComponentDependencySpecificationInput, dfe: DataFetchingEnvironment
+    ): IntraComponentDependencySpecification {
+        return intraComponentDependencySpecificationService.updateIntraComponentDependencySpecification(
+            dfe.gropiusAuthorizationContext, input
+        )
+    }
+
+    @GraphQLDescription(
+        """Deletes the specified IntraComponentDependencySpecification, requires ADMIN on the Component associated with the 
+        IntraComponentDependencySpecification to delete."""
+    )
+    @AutoPayloadType("The id of the deleted IntraComponentDependencySpecification")
+    suspend fun deleteIntraComponentDependencySpecification(
+        @GraphQLDescription("Defines which IntraComponentDependencySpecification to delete")
+        input: DeleteNodeInput, dfe: DataFetchingEnvironment
+    ): ID {
+        intraComponentDependencySpecificationService.deleteIntraComponentDependencySpecification(
+            dfe.gropiusAuthorizationContext, input
+        )
+        return input.id
+    }
+
+}
```

Comment:
Would recommend into splitting into `Component`/`Project`, `Interface`, SEFS, `IMS`. ... specific files

Reply:
I'm not sure about this  
I don't really want another arbitrary grouping of domain classes (besides the general architecture, issue, template, ... grouping) 
Especially, as this new grouping would be represented nowhere else in the codebase or domain model
If we want to split it into smaller files, I would probably create one foreach Service